compiler/rustc_borrowck/src/diagnostics/region_name.rs

```diff
@@ -179,6 +179,25 @@ impl RegionName {
             RegionNameSource::Static => {}
         }
     }
+
+    pub(crate) fn highlight_region_name_suggestion(&self, diag: &mut Diagnostic) {
+        match &self.source {
+            RegionNameSource::AnonRegionFromOutput(
+                RegionNameHighlight::Occluded(span, type_name),
+                mir_description,
+            ) => {
+                if type_name[..4] == "Map<".to_string() {
```

Comment:
This probably does not work like it should.

Reply:
That is, we shouldn't be using the typename to tell if we should make this suggestion. It also doesn't work for anything that isn't `Map<` -- lots of other combinators and functions have signatures almost identical to map and should be included in this suggestion.

Reply:
The reason of restricting to map() only (for now) is to exclude those cases when adding `move` is not a correct suggestion. For example, in the test case of #69446 (#72598), adding `move` is not a solution, although the error is the same:
```
error: captured variable cannot escape `FnMut` closure body
  --> /home/ychen/rust.7.22/rust/src/test/ui/async-await/issue-69446-fnmut-capture.rs:19:17
   |
18 |       let mut x = Foo;
   |           ----- variable defined here
19 |       bar(move || async { //~ ERROR captured
   |  _______________-_^
   | |               |
   | |               inferred to be a `FnMut` closure
20 | |         x.foo();
   | |         - variable captured here
21 | |     });
   | |_____^ returns an `async` block that contains a reference to a captured variable, which then escapes the closure body
   |
   = note: `FnMut` closures only have access to their captured variables while they are executing...
   = note: ...therefore, they cannot allow references to captured variables to escape
```
With test case, the signature of typename is "impl Future<Output = ()>".


For `move` keyword suggestion, I thought it is safer to use a "white list" of typename signatures for cases we know will work. such as "Map<" for map(), than use a "black list" to exclude cases we know will not work, like "impl Future" signature for `async`. We can add more signatures to "white list" for cases which can be fixed by adding `move` keyword for sure.

I am not sure if there is any other simple and clean way to tell if a case can be fixed by adding `move` keyword or not.